src/platform/pret_sdl/sdl2.c

```diff
@@ -244,19 +246,27 @@ void VBlankIntrWait(void)
 
         if (!paused || stepOneFrame) {
             double dt = fixedTimestep / timeScale; // TODO: Fix speedup
-            double deltaTime = 0;
 
-            curGameTime = SDL_GetPerformanceCounter();
-            if (stepOneFrame) {
-                deltaTime = dt;
+            // Hack to emulate the behaviour of threaded sdl
+            // it will not add any new values to the accumulator
+            // when a new frame was requested within a frame cycle
+            if (!newFrameRequested) {
```

Comment:
I think we were seeing some performance weirdness because when we "request a new frame" this would add to the accumulator, where as this would not happen on the threaded version since the next frame was "requested" within the accumulator loop.